docs/development/quickstart.md

```diff
@@ -0,0 +1,65 @@
+# Development quickstart
+
+This is quick start guide for developing and building this project from source on a Linux machine.
+
+## Get the source
+
+The canonical development location of this project is on GitHub. You can fetch the full source with history via `git`:
+
+```sh
+git clone https://github.com/coreos/zincati.git
+cd zincati
+```
+
+It is recommend to fork a copy of the project to your own GitHub account, and add it as an additional remote:
+
+```sh
+git remote add my-fork git@github.com:<YOURUSER>/zincati.git
+```
+
+## Install Rust toolchain
+
+This project is written in Rust, and requires a stable toolchain to build. Additionally, `clippy` and `rustfmt` are used by CI jobs to ensure that patches are properly formatted and linted.
+
+You can obtain a Rust toolchain via many distribution methods, but the simplest way is via [rustup](https://rustup.rs/):
+
+```sh
+rustup component add clippy
+rustup component add rustfmt
+rustup install stable
+```
+
+## Build and test
+
+Building and testing is handled via `cargo` and `make`:
+
+```sh
+make build
+make check
+```
+
+If you prefer running builds in a containerized environment, you can use the provided `Dockerfile`:
+
+```sh
+docker build -f contrib/Dockerfile.dev -t coreos/zincati:dev .
```

Comment:
There's also the cosa buildroot image which has cosa + build deps we could link to here. This is actually what CI here itself uses: https://github.com/coreos/zincati/blob/9d5d5ebe941151aff1d1b5efbbb816553de94f9b/.cci.jenkinsfile#L3 -> https://github.com/coreos/coreos-ci-lib/blob/76bbd3d46503dea956c59eb02a079302de0d3abb/vars/cosaPod.groovy#L8.

Nice to have a minimal image as well though given that the buildroot image is massive.